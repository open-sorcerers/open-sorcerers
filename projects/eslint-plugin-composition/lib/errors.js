const PREFER_SELECTOR =
  'Use a selector function instead of a literal with nested property access.';
const preferNamedSelector = named =>
  PREFER_SELECTOR.replace(/a literal/, '"' + named + '"');

module.exports = { PREFER_SELECTOR, preferNamedSelector };
