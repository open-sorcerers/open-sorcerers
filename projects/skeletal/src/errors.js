import { curry } from "ramda"
import { austereStack, deepfreeze } from "./utils"
import { nameVersion } from "./instance"

export const error = curry((ns, message, data) => {
  const name = `${nameVersion()}::${ns}`
  const e = new Error(message)
  e.name = name
  e.data = data
  return austereStack(e)
})
export const ERROR = deepfreeze({
  EXPECTED_NAME_AND_MORE: error(
    `pattern`,
    `Expected pattern to have {name, prompts, actions} properties.`
  ),
  INCOMPLETE_ACTION: error(
    `render`,
    `Expected action to have {type, path, template} properties.`
  )
})
