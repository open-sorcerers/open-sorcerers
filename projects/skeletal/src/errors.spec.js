import { error, ERROR } from "./errors"

test("error", () => {
  const output = error('spectest', 'This is a spectesterror', {cool: 'true'})
  expect(output).toMatchSnapshot()
})
test("ERROR", () => {
  expect(!!ERROR).toBeTruthy()
})
