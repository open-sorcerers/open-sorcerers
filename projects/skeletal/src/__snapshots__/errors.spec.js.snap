// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error 1`] = `[[1mskeletal0.0.6[22m::spectest: This is a spectesterror]`;
